libs/packages/components/src/lib/common-navigation-model/common-navigation-model.ts

```diff
@@ -0,0 +1,32 @@
+export interface INavigationLink {
+    /**
+     * Text to be displayed in the link
+     */
+    text: string;
+
+    /**
+     * Navigation Route 
+     */
+    route: string;
+
+    /**
+     * Internal Angualr Routes, External HREF, EVENT: event on parent component
+     */
+    mode: NavigationMode;
+}
```

Comment:
We should add **target** as an option for internal and external links. There will likely be footer links that will want to target="_blank".

Reply:
Let us discuss this and we could add it in later. Let me get this overhaul done first. How do routes work with blank? I have never tried.

Reply:
So a routerLink and target="_blank" will just open up a new copy of the application at the target route. The developer who uses this just needs to understand that the new tab will be bootstrapped from scratch and will need to be stateless.

Reply:
So normally there is a visual indicator if something opens in a new window. Lets ask about the icon and positioning for it. Lets hold off.